### PR TITLE
FEAT: Updated Command handling

### PR DESCRIPTION
**Pull request by MagicShel**
So this is how I would do command handling. I don't know if you'll hate this or maybe incorporate some of the ideas. The reason for doing this is so that there isn't the possibility of a mismatch between commands that are registered and commands that exist.

### CODE REVIEW

File: src/main/java/es/thalesalv/chatrpg/application/service/commands/UnsetCommandService.java
Comment (thaalesalves) [inline src/main/java/es/thalesalv/chatrpg/application/service/commands/UnsetCommandService.java]: This command should be part of the larger command suites too, I forgot to move it there. As set isn't a thing anymore, this should follow suit.

File: src/main/java/es/thalesalv/chatrpg/application/service/commands/chconf/ChConfCommandService.java
Comment (thaalesalves) [inline src/main/java/es/thalesalv/chatrpg/application/service/commands/chconf/ChConfCommandService.java]: My issue is with this. Now we need an extra class for each command suite. This isn't a complaint, just a comment. This won't really solve the issue between registered and existing commands, just moved the issue to a new class that didn't exist before. Can't this specific bit be done in the global command handler so that it knows exactly which class to call based solely on the class name? I know this works the same way, but looks a bit over-engineered.

File: src/main/java/es/thalesalv/chatrpg/application/service/commands/lorebook/LorebookCommandService.java
Comment (thaalesalves) [inline src/main/java/es/thalesalv/chatrpg/application/service/commands/lorebook/LorebookCommandService.java]: Something I think we should change in the sake of naming: the interface is probably better called `DiscordInteraction` as it covers all interaction types and not just commands.

File: src/main/java/es/thalesalv/chatrpg/application/service/commands/lorebook/LorebookEditHandler.java
Comment (thaalesalves) [inline src/main/java/es/thalesalv/chatrpg/application/service/commands/lorebook/LorebookEditHandler.java]: There are specific exceptions for each not found case. This also needs a more complete message. This exception isn't handled either, so the user won't have any response. Throwing `LorebookEntryNotFoundException` with a specific message that gets shown to the user is the best way here.

File: src/main/java/es/thalesalv/chatrpg/application/service/commands/world/WorldSetHandler.java
Comment (thaalesalves) [inline src/main/java/es/thalesalv/chatrpg/application/service/commands/world/WorldSetHandler.java:57]: Yeah, I like this one. Not really much of a reason for two catches if it'll just show the exception message.

File: src/main/java/es/thalesalv/chatrpg/application/service/commands/lorebook/LorebookCommandService.java
Comment (thaalesalves) [inline src/main/java/es/thalesalv/chatrpg/application/service/commands/lorebook/LorebookCommandService.java]: I like this method here because it forces to actually create the command so that issue of forgetting to register the command doesn't happen. Very good one.

File: src/main/java/es/thalesalv/chatrpg/application/service/commands/world/WorldCommandService.java
Comment (thaalesalves) [inline src/main/java/es/thalesalv/chatrpg/application/service/commands/world/WorldCommandService.java]: Logger methods have interpolation, using it makes it cleaner

File: src/main/java/es/thalesalv/chatrpg/adapters/discord/listener/InteractionListener.java
Comment (thaalesalves) [inline src/main/java/es/thalesalv/chatrpg/adapters/discord/listener/InteractionListener.java]: Don't understand this bit here. Why is it verifying the `helpService` specifically? Is it because it has the list of commands? If so, why is there `byName`? Should the `helpService` have all commands registered by default, so it would never miss commands?

File: src/main/java/es/thalesalv/chatrpg/application/service/BotCommands.java
Comment (thaalesalves) [inline src/main/java/es/thalesalv/chatrpg/application/service/BotCommands.java]: The question I asked about helpService in the other thread extends here. When are each of these methods used?

**Comment by thaalesalves** [comment]
Overall I like the refactor, but there are some things there that I'd like to understand a bit better. About maintainability. Right now, all we need to create a new command is add the proper class and register the command because we have a factory that injects the correct bean automatically. That factory was removed and 2 or 3 new classes were added in its place. That is the part I don't really understand. Idk if I'm getting this wrong, but it seems to add steps to maintainability and looks a bit over-engineered. 

How does adding new commands work with this version of command handling? Why is a specific service for each super-command necessary instead of a factory that automatically retrieves the command class for execution? If I forget to add the new sub-command to the switch case in the larger command service, the command won't work, so the problem with commands that were not registered isn't really solved, it was just moved. Did I get this wrong?

**Comment by MagicShel** [comment]
The thinking is a little backwards here. The "super" command isn't a central part of the design, rather in the case of complex logic, the command is divided up to make the classes smaller and easier to maintain. That's actually not the case for the majority of the commands, just the big three.

From an architectural standpoint, you can kinda tell because these handlers don't share a common interface so they are more ad hoc than architectural components, although documentation would make that clearer.

File: src/main/java/es/thalesalv/chatrpg/application/service/commands/lorebook/LorebookEditHandler.java
Comment (MagicShel) [reply src/main/java/es/thalesalv/chatrpg/application/service/commands/lorebook/LorebookEditHandler.java]: Using get without isPresent is code smell. In fact using get at all is almost always code smell.

File: src/main/java/es/thalesalv/chatrpg/adapters/discord/listener/InteractionListener.java
Comment (MagicShel) [reply src/main/java/es/thalesalv/chatrpg/adapters/discord/listener/InteractionListener.java]: It's because of circular dependency. Help needs to have all the commands, but it also needs to be in the commands. No great way to handle this excerpt to combine the help handler with the command strategy factory.

File: src/main/java/es/thalesalv/chatrpg/application/service/commands/lorebook/LorebookEditHandler.java
Comment (thaalesalves) [reply src/main/java/es/thalesalv/chatrpg/application/service/commands/lorebook/LorebookEditHandler.java]: Oh, the comment wasn't about getting rid of get, it was about the exception thrown.

**Comment by thaalesalves** [comment]
> The thinking is a little backwards here. The "super" command isn't a central part of the design, rather in the case of complex logic, the command is divided up to make the classes smaller and easier to maintain. That's actually not the case for the majority of the commands, just the big three.
> 
> From an architectural standpoint, you can kinda tell because these handlers don't share a common interface so they are more ad hoc than architectural components, although documentation would make that clearer.

That makes sense. Well, I like the general idea here, only the things we spoke about that I'd need to correct. I'll approve the PR and once those are done we can merge it.

File: src/main/java/es/thalesalv/chatrpg/application/service/commands/chconf/ChConfCommandService.java
Comment (MagicShel) [reply src/main/java/es/thalesalv/chatrpg/application/service/commands/chconf/ChConfCommandService.java]: @thaalesalves  Are you expecting me to resolve this? Because undoing this really undoes this whole PR. We go from InteractionHandlers which can register themselves to ActionHandlers that can't because they share a command name.

Comment (thaalesalves) [reply src/main/java/es/thalesalv/chatrpg/application/service/commands/chconf/ChConfCommandService.java]: Nah, just the rest.

Comment (thaalesalves) [reply src/main/java/es/thalesalv/chatrpg/application/service/commands/chconf/ChConfCommandService.java]: You can ignore this one.

File: src/main/java/es/thalesalv/chatrpg/application/service/commands/lorebook/LorebookCommandService.java
Comment (thaalesalves) [reply src/main/java/es/thalesalv/chatrpg/application/service/commands/lorebook/LorebookCommandService.java]: This one that could be done for this PR.

File: src/main/java/es/thalesalv/chatrpg/application/service/commands/lorebook/LorebookEditHandler.java
Comment (thaalesalves) [reply src/main/java/es/thalesalv/chatrpg/application/service/commands/lorebook/LorebookEditHandler.java]: This one that could be done for this PR.

File: src/main/java/es/thalesalv/chatrpg/application/service/commands/world/WorldCommandService.java
Comment (thaalesalves) [reply src/main/java/es/thalesalv/chatrpg/application/service/commands/world/WorldCommandService.java]: This one that could be done for this PR.

**Review by thaalesalves — APPROVED** [review]

**Pull request merged**
